yarn-project/simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -78,75 +91,32 @@ export class Call extends Instruction {
 
     context.machineState.incrementPc();
   }
-}
-
-export class StaticCall extends Instruction {
-  static type: string = 'STATICCALL';
-  static readonly opcode: Opcode = Opcode.STATICCALL;
-  // Informs (de)serialization. See Instruction.deserialize.
-  static readonly wireFormat: OperandType[] = [
-    OperandType.UINT8,
-    OperandType.UINT8,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    OperandType.UINT32,
-    /* temporary function selector */
-    OperandType.UINT32,
-  ];
 
-  constructor(
-    private indirect: number,
-    private _gasOffset: number /* Unused due to no formal gas implementation at this moment */,
-    private addrOffset: number,
-    private argsOffset: number,
-    private argsSize: number,
-    private retOffset: number,
-    private retSize: number,
-    private successOffset: number,
-    private temporaryFunctionSelectorOffset: number,
-  ) {
-    super();
+  public get type(): 'CALL' | 'STATICCALL' {
+    const type = super.type;
+    if (type !== 'CALL' && type !== 'STATICCALL') {
+      throw new Error(`Invalid type for ExternalCall instruction: ${type}`);
+    }
+    return type;
   }
 
-  async execute(context: AvmContext): Promise<void> {
-    const [_gasOffset, addrOffset, argsOffset, retOffset, successOffset] = Addressing.fromWire(this.indirect).resolve(
-      [this._gasOffset, this.addrOffset, this.argsOffset, this.retOffset, this.successOffset],
-      context.machineState.memory,
-    );
-
-    const callAddress = context.machineState.memory.get(addrOffset);
-    const calldata = context.machineState.memory.getSlice(argsOffset, this.argsSize).map(f => f.toFr());
-    const functionSelector = context.machineState.memory.getAs<Field>(this.temporaryFunctionSelectorOffset).toFr();
-
-    const nestedContext = context.createNestedContractStaticCallContext(
-      callAddress.toFr(),
-      calldata,
-      FunctionSelector.fromField(functionSelector),
-    );
-
-    const nestedCallResults = await new AvmSimulator(nestedContext).execute();
-    const success = !nestedCallResults.reverted;
-
-    // We only take as much data as was specified in the return size -> TODO: should we be reverting here
-    const returnData = nestedCallResults.output.slice(0, this.retSize);
-    const convertedReturnData = returnData.map(f => new Field(f));
+  protected execute(_context: AvmContext): Promise<void> {
+    throw new Error(`Unimplemented`);
+  }
 
-    // Write our return data into memory
-    context.machineState.memory.set(successOffset, new Uint8(success ? 1 : 0));
-    context.machineState.memory.setSlice(retOffset, convertedReturnData);
+  protected gasCost(): Gas {
+    throw new Error(`Unimplemented`);
```

Comment:
Hmm, this will stay unimplemented by the derived classes because you only know the cost at execution time, right? Could you add a comment?